### PR TITLE
Week five optimizing optional attendees

### PR DESCRIPTION
**Pull request by EmmaWenzel**
All changes to the javascript and html files were given to us by the intern team to fix a bug

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 public final class FindMeetingQuery {
 
   /** Removes TimeRanges from times collection that are unavailable for the given attendees */
-  private ArrayList<TimeRange> removeConflictingTimes(ArrayList<TimeRange> times, Collection<String> attendees, Collection<Event> events, long duration){
+  private ArrayList<TimeRange> removeConflictingTimes(ArrayList<TimeRange> times, Collection<String> attendees, Collection<Event> events, long duration, boolean optimizing){
```
Comment (devonwasson) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:31]: Booleans should typically be named something like isX where X is the thing you're checking for (https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)

So maybe something like isOptiminzingMeetingTimes would be good. it's a bit wordy but it tells the reader exactly what you mean with no extra effort required.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -113,12 +115,51 @@
...
+    // that allow mandatory attendees and the greatest possible number of optional attendees to attend.
     if(times.isEmpty() && !(attendees.isEmpty())){
+
         times = timesBeforeOptional;
+
+        // breaks times into smaller peices that are available for different optional attendees
+        times = removeConflictingTimes(times, optionalAttendees, events, request.getDuration(), true);
+
+        int lowestConflicts = optionalAttendees.size() - 1;
+        ArrayList<TimeRange> mostPeopleAvailable = new ArrayList<>();
```
Comment (devonwasson) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:131]: this variable name and its type don't seem to match. maybe something like bestTimeRanges

**Review by RegExrTech — CHANGES_REQUESTED** [review]
tests look good! logic seems pretty solid as well, and if the tests pass then looks like you nailed it! just a few nits about naming conventions but otherwise, good to go!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (EmmaWenzel) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Got it, thanks!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (EmmaWenzel) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Alright, cool I changed that.

**Review by RegExrTech — APPROVED** [review]
